bot/exts/valentines/valentine_zodiac.py

```diff
@@ -19,36 +22,117 @@ class ValentineZodiac(commands.Cog):
+        """Gives informative zodiac embed."""
+        zodiac = zodiac.capitalize()
+        embed = discord.Embed()
+        embed.color = Colours.pink
+        if zodiac in self.zodiac_fact:
+            log.info("Making zodiac embed.")
+            embed.title = f"__{zodiac}__"
+            embed.description = self.zodiac_fact[zodiac]["About"]
+            embed.add_field(name='__Full form__', value=self.zodiac_fact[zodiac]["full_form"], inline=False)
+            embed.add_field(name='__Motto__', value=self.zodiac_fact[zodiac]["Motto"], inline=False)
+            embed.add_field(name='__Strengths__', value=self.zodiac_fact[zodiac]["Strengths"], inline=False)
+            embed.add_field(name='__Weaknesses__', value=self.zodiac_fact[zodiac]["Weaknesses"], inline=False)
+            embed.set_thumbnail(url=self.zodiac_fact[zodiac]["url"])
+        else:
+            embed = self.error(zodiac)
+        log.info("Zodiac embed ready.")
+        return embed
+
+    def zodiac_date_verifier(self, query_datetime: date) -> str:
+        """Returns zodiac sign by checking month and date."""
+        for zodiac_name, zodiac_data in self.zodiac_fact.items():
+            if zodiac_data["start_at"] <= query_datetime <= zodiac_data["end_at"]:
+                log.info("Zodiac name sent.")
+                return zodiac_name
+
+    @commands.group(name='zodiac', invoke_without_command=True)
+    async def zodiac(self, ctx: commands.Context, zodiac_sign: str) -> None:
+        """Provides information about zodiac sign by taking zodiac sign name as input."""
+        final_embed = self.zodiac_build_embed(zodiac_sign)
+        await ctx.send(embed=final_embed)
+        log.info("Embed successfully sent.")
+
+    @zodiac.command(name="date")
+    async def date_and_month(self, ctx: commands.Context, query_date: int, month: Union[int, str]) -> None:
+        """Provides information about zodiac sign by taking month and date as input."""
+        if isinstance(month, str):
+            month = month.capitalize()
+            try:
+                month = list(calendar.month_abbr).index(month[:3])
+            except ValueError:
+                await ctx.send(f"Sorry, but `{month}` is not a valid month name.")
+                return
+        if (month == 1 and (1 <= query_date <= 19)) or (month == 12 and (22 <= query_date <= 31)):
+            zodiac = "capricorn"
+            final_embed = self.zodiac_build_embed(zodiac)
+        else:
+            try:
+                zodiac_sign_based_on_month_and_date = self.zodiac_date_verifier(date(2020, month, query_date))
+                log.info("zodiac sign based on month and date received.")
+            except ValueError as e:
+                final_embed = discord.Embed()
+                final_embed.color = Colours.pink
+                final_embed.description = f"Zodiac sign is not found because, {e}"
+                log.info(f"error caused due to: {e}.")
```

Comment:
```suggestion
                log.info(f"Error in "zodiac date" command:\n{e}.")
```
Location is very helpful when debuging